### PR TITLE
infra: use terser on tf_js_binary

### PR DESCRIPTION
**Pull request by stephanwlee**
Currently, TensorBoard only uses rollup to bundle and that is it. While
rollup provides some optimization such as tree shaking, it does not do
any variable mangling and useless thing stripping. This change adds
terse to tf_js_bianry which we use for production JS binary.

Before: index.js = 4.0MB gzipped. 18.7MB uncompressed.
After: index.js = 2.3MB gzipped. 9.5MB uncompressed.

### CODE REVIEW

**Review by wchargin — APPROVED** [review]
Nice; thanks! Worth a test sync?

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "@bazel/karma": "^2.3.0",
     "@bazel/protractor": "^2.3.0",
     "@bazel/rollup": "^2.3.0",
-    "@bazel/terser": "^2.3.0",
+    "@bazel/terser": "^2.3.2",
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply package.json]: Artifact of `yarn add`. Reverted.

File: tensorboard/defs/BUILD [hunk 2]
```diff
@@ -37,6 +37,7 @@ tb_proto_library(
 
 exports_files([
     "rollup_config.js",
+    "terser_option.json",
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/defs/BUILD]: Changed to terser_config.json

**Comment by stephanwlee** [comment]
Tested the sync and everything looked good afterwards as expected.

**Pull request merged**